lms/validation/_oauth.py

```diff
@@ -0,0 +1,6 @@
+"""Validation for OAuth views."""
+from webargs import fields
+
+
+#: Arguments for the canvas_oauth_callback() view.
+CANVAS_OAUTH_CALLBACK_ARGS = {"state": fields.Str(required=True)}
```

Comment:
This is the actual validation schema. You can write these as full on [marshmallow](https://marshmallow.readthedocs.io/) schema classes (marshmallow is the validation lib that webargs wraps), but webargs also provides this easy dict format.

This dict fits on one line because we're only validating one field so far but normally these would be larger multiline dicts.

The idea is that this `_oauth.py` module will grow to contain all the OAuth-related schemas, and more `_foo.py` modules will be added to contain more schemas.